Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCiscoASA.yaml

```diff
@@ -0,0 +1,576 @@
+                                 AliasDeviceVersion = DeviceVersion,
+                                 AliasDeviceAction = DeviceAction,
+                                 NetworkRuleName = DeviceCustomString2,
+                                 SrcIpAddr = SourceIP,
+                                 SrcPortNumber = SourcePort,
+                                 DstIpAddr = DestinationIP, 
+                                 DstPortNumber = DestinationPort
+                | project-away Activity, AdditionalExtensions, ApplicationProtocol, Device*, Destination*, EndTime, ExternalID, File*, Flex*, IndicatorThreatType, Malicious*, Old*, Process*, ReceiptTime, ReceivedBytes, Remote*, Report*, Request*, Source*, SentBytes, StartTime, TenantId, Threat*;
+      let unparsedData = allLogs
+                | where isempty(SourceIP)
+                | project-rename AliasDeviceEventClassID = DeviceEventClassID
+                | project-away Activity, AdditionalExtensions, ApplicationProtocol, Device*, Destination*, EndTime, ExternalID, File*, Flex*, IndicatorThreatType, Malicious*, Old*, Process*, ReceiptTime, ReceivedBytes, Remote*, Report*, Request*, Source*, SentBytes, StartTime, TenantId, Threat*;
+      let all_106001_alike = parsedData
+                | where AliasDeviceEventClassID in ("106001", "106006", "106015", "106016", "106021", "106022") 
+                | parse Message with * " interface " DstInterfaceName;
+      let all_106010_alike = parsedData
+                | where AliasDeviceEventClassID in ("106010", "106014")
+                | parse Message with * " src "  SrcInterfaceName ":" * " dst " DstInterfaceName ":" * "(type " NetworkIcmpType ", code " NetworkIcmpCode:int ")";
+      let all_106018 = parsedData
+                | where AliasDeviceEventClassID == "106018"
+                | parse Message with * " packet type " NetworkIcmpType " " * "list " NetworkRuleName " " *;
+      let all_106023 = parsedData
+                | where AliasDeviceEventClassID == "106023" and not(Message has "protocol 41")
+                | parse Message with * " src "  SrcInterfaceName ":" * " dst " DstInterfaceName ":" * ' by access-group "' NetworkRuleName '" '  *
+                | parse Message with * "(type " NetworkIcmpType ", code " NetworkIcmpCode:int ")" *;
+      let all_106023_unparsed = unparsedData
+                | where AliasDeviceEventClassID == "106023" and not(Message has "protocol 41")
+                | parse Message with * ":" DeviceAction " " Protocol " src " SrcInterfaceName ":" SrcIpAddrAndPort "(" SrcUsername ") dst " DstInterfaceName ":" DstIpAddrAndPort " " NetworkIcmpInfo 'by access-group "' NetworkRuleName '" [' * "]"
+                | parse NetworkIcmpInfo with "(type " NetworkIcmpType ", code " NetworkIcmpCode:int ") "
+                | extend SrcIpAddrAndPort = split(SrcIpAddrAndPort,"/"), DstIpAddrAndPort = split(DstIpAddrAndPort,"/")
+                | extend SrcIpAddr = tostring(SrcIpAddrAndPort[0]),
```

Comment:
Missed asking on the ASIM version: why don't IP addresses and ports work int he "parse" statement itself? Is the port optional?

Reply:
Yes Ofer, you are right. We have port value as optional.